Model/FilePayload.php

```diff
@@ -238,11 +247,37 @@ public function setAttribution($attribution)
     }
 
     /**
-     * @return \MauticPlugin\MauticContactClientBundle\Entity\QueueRepository
+     * @return QueueRepository
      */
     private function getQueueRepository()
     {
-        return $this->em->getRepository('MauticContactClientBundle:Queue');
+        /** @var QueueRepository $repo */
+        $repo = $this->getEntityManager()->getRepository('MauticContactClientBundle:Queue');
+
+        return $repo;
+    }
+
+    /**
+     * Shore up EntityManager loading, in case there is a flaw in a plugin or campaign handling.
+     *
+     * @return EntityManager
+     */
+    private function getEntityManager()
+    {
+        try {
+            if ($this->em && !$this->em->isOpen()) {
+                $this->em = $this->em->create(
+                    $this->em->getConnection(),
+                    $this->em->getConfiguration(),
+                    $this->em->getEventManager()
+                );
+                // $this->logger->error('ContactClient: EntityManager was closed.');
```

Comment:
Should this be uncommented as well?

Also looks like this should be a trait, or some kind of service perhaps....

Reply:
I should have made this a trait, but a trait would then need the logger service... I'm hoping that we don't actually need any of these getEntityManager() methods and we can drop them in the next release. I'm just trying to stop some fires.